crates/build/re_types_builder/src/codegen/python/mod.rs

```diff
@@ -2591,6 +2593,100 @@ fn quote_partial_update_methods(reporter: &Reporter, obj: &Object, objects: &Obj
     ))
 }
 
+fn quote_columnar_methods(reporter: &Reporter, obj: &Object, objects: &Objects) -> String {
+    let parameters = obj
+        .fields
+        .iter()
+        .filter_map(|field| {
+            let mut field = field.make_plural()?;
+            field.is_nullable = true;
+            Some(quote_init_parameter_from_field(
+                &field,
+                objects,
+                &obj.fqname,
+            ))
+        })
+        .collect_vec()
+        .join(",\n");
+    let parameters = indent::indent_by(8, parameters);
+
+    let init_args = obj
+        .fields
+        .iter()
+        .map(|field| {
+            let field_name = field.snake_case_name();
+            format!("{field_name}={field_name}")
+        })
+        .collect_vec()
+        .join(",\n");
+    let init_args = indent::indent_by(12, init_args);
+
+    let parameter_docs = compute_init_parameter_docs(reporter, obj, objects);
+    let doc = unindent(
+        "\
+        Partitions the component data into multiple sub-batches.
+
+        This makes it possible to use `rr.send_columns` to send columnar data directly into Rerun.
+
+        If specified, `_lengths` must sum to the total length of the component batch.
```

Comment:
I don't think the various LSPs appreciate the leading `_` :grimacing: I need to put it at the end instead.